molgenis/molgenis-emx2
# fix: when using subclasses should not be able to remove data when it is referred to

closes #3144 

to my amazing we didn't use primary keys, but unique, and also no foreign key from subclass to superclass tables.
this PR fixes both (and I noticed a test that already noticed the suspicious behaviour causing #3144).

how to test:
- to reproduce issue, you should delete a row in Variable that is used in a mapping or variable value via 'All variables'
- before this fix, you could to this because All Variables was not deleted
- with this fix you get a foreign key error prohibiting this delete as it should be

todo:
- [x] somebody else verify that 3144 indeed is fixed
- [x] enable to rename primary key fields (required some magic to also change in subclass tables)
- [x] added migration script 17 for old schema missing this fkey
